### DISCUSSION TITLE
After hot reloads, out of heap memory occurs

### DISCUSSION THREAD

**Discussion by tarun1992** [vercel/next.js#21737] [Help]
**What version of Next.js are you using?**

10.0.6

**What version of Node.js are you using?**

12

**What browser are you using?**

Chrome

**What operating system are you using?**

Windows 10

**How are you deploying your application?**

Vercel

**Describe the Bug**

I start my application by using 'npm run dev' and after doing some changes(after 10 min or so) - It gives a heap out of memory error.  I am using Next.js for quite some time and never faced such a thing. This happened after I updated my current version 9.4 to 10.0.6(latest). 
Below is a stack trace  -

[24208:0000019A22F30240]  2718746 ms: Mark-sweep 2038.3 (2052.8) -> 2037.6 (2052.6) MB, 2311.2 / 0.1 ms  (average mu = 0.096, current mu = 0.048) allocation failure scavenge might not succeed
[24208:0000019A22F30240]  2718757 ms: Scavenge 2038.4 (2052.6) -> 2038.2 (2052.8) MB, 8.4 / 0.0 ms  (average mu = 0.096, current mu = 0.048) allocation failure
FATAL ERROR: Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory
 1: 00007FF604605EBF napi_wrap+114095
 2: 00007FF6045B0B46 v8::base::CPU::has_sse+66998
 3: 00007FF6045B1946 v8::base::CPU::has_sse+70582
 4: 00007FF604DC6E4E v8::Isolate::ReportExternalAllocationLimitReached+94
 5: 00007FF604DAEF21 v8::SharedArrayBuffer::Externalize+833
 6: 00007FF604C7B18C v8::internal::Heap::EphemeronKeyWriteBarrierFromCode+1436
 7: 00007FF604C863C0 v8::internal::Heap::ProtectUnprotectedMemoryChunks+1312
 8: 00007FF604C82EE4 v8::internal::Heap::PageFlagsAreConsistent+3204
 9: 00007FF604C786E3 v8::internal::Heap::CollectGarbage+1283
10: 00007FF604C76D54 v8::internal::Heap::AddRetainedMap+2452
11: 00007FF604C9809D v8::internal::Factory::NewFillerObject+61
12: 00007FF6049FE1E1 v8::internal::interpreter::JumpTableTargetOffsets::iterator::operator=+1665
13: 00007FF60521B3BD v8::internal::SetupIsolateDelegate::SetupHeap+546637
14: 00000294B8AE8587
npm ERR! code ELIFECYCLE
npm ERR! errno 134
npm ERR! bikewalevlog@1.0.0 dev: `next`
npm ERR! Exit status 134
npm ERR!

**Expected Behavior**

The out of heap error should be addressed.

**To Reproduce**

This is something that happens randomly on the local system. I tried reproducing on my local and it occurs randomly. I am quite sure though it never occurred before but came after upgrading.

**Comment by timneutkens** [comment]
Converting to discussion as this is not actionable.

**Comment by city41** [comment]
This also happens to me. I have two NextJS projects which both exhibit this. It seems to mostly happen after I make a change in a `scss` file. I've also noticed sass related changes can take a very long time (on the order of minutes). But I don't think this is exclusive to sass changes, as after a while, hot reloads will eventually cause node to crash with `Allocation failed -  JavaScript heap out of memory`.

I am using NextJS 10.0.5, and node 12.20.0. I've tried several versions of node and all exhibit the same behavior.

My NextJS projects are here

https://github.com/city41/mattgreer.dev
https://github.com/city41/stepcharts

I understand just pointing to two giant projects is not very useful. If I get some free time I will attempt to create a small example that repros this issue.
